pyomo/contrib/parmest/parmest.py

```diff
@@ -613,18 +1181,234 @@ def _Q_opt(
+                if all(item is None for item in meas_error):
+                    measurement_var = sse / (
+                        n - l
+                    )  # estimate of the measurement variance
+                    if cov_method == CovarianceMethodLib.reduced_hessian:
+                        cov = (
+                            2 * measurement_var * self.inv_red_hes
+                        )  # covariance matrix
+                        cov = pd.DataFrame(
+                            cov,
+                            index=self.estimated_theta.keys(),
+                            columns=self.estimated_theta.keys(),
+                        )
+                    elif (
+                        cov_method == CovarianceMethodLib.finite_difference
+                        or cov_method
+                        == CovarianceMethodLib.automatic_differentiation_kaug
+                    ):
+                        cov = compute_covariance_matrix(
+                            self.exp_list,
+                            method,
+                            theta_vals=self.estimated_theta,
+                            solver=solver,
+                            step=step,
+                            tee=self.tee,
+                            estimated_var=measurement_var,
+                        )
+                    else:
+                        raise NotImplementedError(
+                            'Only "finite_difference", "reduced_hessian", and '
+                            '"automatic_differentiation_kaug" methods are supported.'
+                        )
+                elif all(item is not None for item in meas_error):
+                    if cov_method == CovarianceMethodLib.reduced_hessian:
+                        cov = 2 * (meas_error[0] ** 2) * self.inv_red_hes
+                        cov = pd.DataFrame(
+                            cov,
+                            index=self.estimated_theta.keys(),
+                            columns=self.estimated_theta.keys(),
+                        )
+                    elif (
+                        cov_method == CovarianceMethodLib.finite_difference
+                        or cov_method
+                        == CovarianceMethodLib.automatic_differentiation_kaug
+                    ):
+                        cov = compute_covariance_matrix(
+                            self.exp_list,
+                            method,
+                            theta_vals=self.estimated_theta,
+                            solver=solver,
+                            step=step,
+                            tee=self.tee,
+                        )
+                    else:
+                        raise NotImplementedError(
+                            'Only "finite_difference", "reduced_hessian", and '
+                            '"automatic_differentiation_kaug" methods are supported.'
+                        )
```

Comment:
This is unreachable code. You already check in lines 1274-1280 whether or not a supported method was supplied, and it raises an error if not. So this is redundant.